### PR TITLE
Fix `ldelem[a]/stelem/cpobj/ldobj/stobj <pointer type>` import

### LINKED ISSUE #71636: `ldelem[a]` importation treats pointers as `object`s
Reproduction:
```cs
[MethodImpl(MethodImplOptions.NoInlining)]
private static ref int* Problem(int*[] a)
{
    return ref a[0];
}
```
Expected result: the address is loaded directly.

Actual result: a helper is used:
```scala
STMT00000 ( 0x000[E-] ... ??? )
               [000004] --CXG------                         *  RETURN    byref 
               [000003] --CXG------                         \--*  CALL help byref  HELPER.CORINFO_HELP_LDELEMA_REF
               [000000] ----------- arg0                       +--*  LCL_VAR   ref    V00 arg0         
               [000001] ----------- arg1                       +--*  CNS_INT   int    0
               [000002] H---------- arg2                       \--*  CNS_INT(h) long   0x7ffc180de192 class
```
Cause (likely): #50749.

### PR DESCRIPTION
**Pull request by SingleAccretion**
To treat pointers as integers and not `object`s.

Fixes #71636.

### CODE REVIEW

File: src/coreclr/jit/importer.cpp [hunk 1]
```diff
@@ -13675,53 +13675,37 @@ void Compiler::impImportBlockCode(BasicBlock* block)
...
+                GenTree* arr   = impPopStack().val;
+                op1            = gtNewHelperCallNode(CORINFO_HELP_LDELEMA_REF, TYP_BYREF, arr, index, type);
 
                 impPushOnStack(op1, tiRetVal);
-                break;
+            }
+            break;
 
             // ldelem for reference and value types
             case CEE_LDELEM:
```
Comment (jkotas) [inline src/coreclr/jit/importer.cpp:13711]: `CEE_STELEM` has the same problem

File: src/coreclr/jit/importer.cpp [hunk 2]
```diff
@@ -13675,53 +13675,37 @@ void Compiler::impImportBlockCode(BasicBlock* block)
...
 
                 JITDUMP(" %08X", resolvedToken.token);
 
                 ldelemClsHnd = resolvedToken.hClass;
+                lclTyp       = JITtype2varType(info.compCompHnd->asCorInfoType(ldelemClsHnd));
 
-                // If it's a value class array we just do a simple address-of
-                if (eeIsValueClass(ldelemClsHnd))
+                // If it's a value class / pointer array, or a readonly access, we don't need a type check.
+                if ((lclTyp != TYP_REF) || ((prefixFlags & PREFIX_READONLY) != 0))
```
Comment (jkotas) [inline src/coreclr/jit/importer.cpp:13689]: We can also apply `impCanSkipCovariantStoreCheck` to skip the helper, similar to how we do it in `CEE_STELEM_REF`.

For example, we can pass modify `impCanSkipCovariantStoreCheck` to treat `NULL` `value` as the ldelema case.

File: src/coreclr/jit/importer.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/importer.cpp:13708]: Indeed, and a couple other cases. I've fixed all of them.

**Review by jkotas — APPROVED** [review]

**Review by EgorBo — APPROVED** [review]

**Comment by EgorBo** [comment]
Thanks!

**Pull request merged**